Forwarding tracing headers when invoking lambdas.

tomgco opened the pull request:
This will forward the trace-request-id to the invoked lambda if present in the original event.

This is being patched in for now to maintain backwards compatibility, however eventually I would like to move all of our Payloads to a format like:

``` js
{
    body: { ... },
    headers: {
        'trace-request-id': 'some-id'
    }
}
```

However as it currently the headers attribute is being re-assigned to a new object.

> **Note:** This assumes that we are not currently using the attribute `headers` on the Payload, if we are then we have two options. Either go straight to the (body / headers) change noted above, causing a lot of changes throughout many services. Or we can add the logic to check for the property and then extend it, if it already exists.

tomgco committed:
Forwarding headers when invoking lambdas.

tomgco committed:
Fixing grammar in comments.

tomgco commented:
@nelsonic @jruts Would you mind casting your eyes over this?

nelsonic commented:
@tomgco looks good. are you making any further changes 
e.g. usage example in readme? or can we merge?

tomgco committed:
Fixing init arity

tomgco committed:
Simplifying logic of retrieving headers

tomgco commented:
> @tomgco looks good. are you making any further changes 
> e.g. usage example in readme? or can we merge?

@nelsonic I have just rebased and simplified some logic based on the other PR that has merged (and also a documentation fix). Everything around usage is documented sufficiently, however around the architecture I will update https://github.com/numo-labs/inspirational-search-docs with all of the required information.

So LGTM; and ready to merge.

nelsonic commented:
@tomgco cool. agree. merging. 👍
